### PR TITLE
Add back_to_game

### PR DESCRIPTION
**Pull request by 3Mydlo3**
Refactored from preset crashTeleport, originaly created as mission script by Madin.

**When merged this pull request will:**
- Teleport back after crash/disconnect
- Restore loadout

### CODE REVIEW

File: addons/back_to_game/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+- Disable addon functionality
+- Teleport to group leader
+- Teleportation to previous vehicle
+- Both combined allow teleportation to leader's vehicle
+
+## Authors
+
+- [3Mydlo3](https://github.com/3Mydlo3)
+- Madin5
```
Comment (veteran29) [inline addons/back_to_game/README.md]:
```suggestion
- [Madin5](https://github.com/Madin5)
```

File: addons/back_to_game/functions/fnc_dialogReject.sqf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Example:
+ * call afm_back_to_game_fnc_dialogReject
+ *
+ * Public: No
+ */
+
+[{
+    GVAR(savegameData) = [];
+}] call CBA_fnc_waitAndExecute;
```
Comment (veteran29) [inline addons/back_to_game/functions/fnc_dialogReject.sqf]: Not sure why is it delayed but if it needs to be delayed use execNextFrame
```suggestion
}] call CBA_fnc_execNextFrame;
```

File: addons/back_to_game/functions/fnc_handleConnected.sqf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Example:
+ * [player, getPlayerUID player] call afm_back_to_game_fnc_handleConnected
+ *
+ * Public: No
+ */
+
+params ["_unit", "_uid"];
+
+if (!isServer) then {
```
Comment (veteran29) [inline addons/back_to_game/functions/fnc_handleConnected.sqf]: No need to fire the event for HCs:
```suggestion
if (!isServer && {hasInterface}) then {
```

Comment (veteran29) [reply addons/back_to_game/functions/fnc_handleConnected.sqf]: Also, shouldn't this be an `exitWith` ?

File: addons/back_to_game/functions/fnc_savePlayerData.sqf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * [player, getPlayerUID player] call afm_back_to_game_fnc_savePlayerData
+ *
+ * Public: No
+ */
+
+params ["_unit", "_uid"];
+
+// Save player's loadout, vehicle and position
+private _loadout = getUnitLoadout _unit;
+private _pos = getposATL _unit;
```
Comment (veteran29) [inline addons/back_to_game/functions/fnc_savePlayerData.sqf]:
```suggestion
private _pos = getPosATL _unit;
```

File: addons/back_to_game/teleportDialog.hpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    class controls {
+        class RscFrame_1058: RscFrame {
+            idc = 1800;
+            x = "0.402031 * safezoneW + safezoneX";
+            y = "0.357 * safezoneH + safezoneY";
+            w = "0.195937 * safezoneW";
+            h = "0.165 * safezoneH";
+        };
+        class RscText_1059: RscText {
```
Comment (veteran29) [inline addons/back_to_game/teleportDialog.hpp]:
```suggestion
        class TeleportText: RscText {
```

File: addons/back_to_game/teleportDialog.hpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            idc = 1000;
+            text = CSTRING(TeleportAsk);
+            x = "0.412344 * safezoneW + safezoneX";
+            y = "0.368 * safezoneH + safezoneY";
+            w = "0.180469 * safezoneW";
+            h = "0.055 * safezoneH";
+        };
+        class RscButton_1060: RscButton {
```
Comment (veteran29) [inline addons/back_to_game/teleportDialog.hpp]:
```suggestion
        class AgreeButton: RscButton {
```

File: addons/back_to_game/teleportDialog.hpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            idc = 1600;
+            text = CSTRING(TeleportAgree);
+            x = "0.407187 * safezoneW + safezoneX";
+            y = "0.445 * safezoneH + safezoneY";
+            w = "0.0670312 * safezoneW";
+            h = "0.066 * safezoneH";
+            action = QUOTE(call FUNC(dialogConfirm); closeDialog 0);
+        };
+        class RscButton_1061: RscButton {
```
Comment (veteran29) [inline addons/back_to_game/teleportDialog.hpp]:
```suggestion
        class RejectButton: RscButton {
```

File: addons/back_to_game/functions/fnc_hasDisconnected.sqf [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+params ["_uid"];
+
+private _save = [_uid] call FUNC(getPlayerData);
+if (isNil "_save") then {
+    false
+} else {
+    true
+};
```
Comment (veteran29) [inline addons/back_to_game/functions/fnc_hasDisconnected.sqf]: ??

```sqf
!isNil {[_uid] call FUNC(getPlayerData)} // return
```

File: addons/back_to_game/functions/fnc_handleConnected.sqf (hunk 3 above)
Comment (3Mydlo3) [reply addons/back_to_game/functions/fnc_handleConnected.sqf]: Yep, it should be with `exitWith`.

File: addons/back_to_game/functions/fnc_hasDisconnected.sqf (hunk 8 above)
Comment (3Mydlo3) [reply addons/back_to_game/functions/fnc_hasDisconnected.sqf]: Yes, I meant exactly that.

**Pull request merged**